jobs/periodic/redact-inactive-accounts/send-inactivity-notification.ts

```diff
@@ -0,0 +1,27 @@
+import * as Notification from '../../../common/notification';
+import { Person } from '../../../common/notification/types';
+import moment from 'moment';
+import { findAllPersons } from './query';
+import { getLogger } from '../../../common/logger/logger';
+
+export const NOTIFY_AFTER_DAYS = 3 * 365 + 9 * 30; // 3 years and 9 months
+
+const logger = getLogger('SendInactivityNotification');
+
+export async function sendInactivityNotification() {
+    logger.info('Start sending inactivity notification');
+
+    const p = await findAllPersons(true, moment().startOf('day').subtract(NOTIFY_AFTER_DAYS, 'days').toDate());
+    logger.info('Sending inactivity notification to users', {
+        pupils: p.pupils.map((p) => p.id),
+        students: p.students.map((p) => p.id),
+        mentors: p.mentors.map((p) => p.id),
+        screener: p.screener.map((p) => p.id),
+    });
+
+    const persons: Person[] = [...p.pupils, ...p.students, ...p.mentors, ...p.screener];
+    const promises = persons.map((person) => Notification.actionTaken(person, 'person_inactivity_reminder', {}));
```

Comment:
better not parallelize this, this might overwhelm the prisma connection pool

Reply:
You could set uniqueId: lastLogin in the context to avoid duplicate notifications (but then the job will still try to notify a user for three months)

Reply:
I think it's better than nothing for now.